linetools/isgm/utils.py

```diff
@@ -463,6 +471,127 @@ def group_coincident_compoments(comp_list, output_type='list'):
     if output_type not in ['list', 'dict', 'dictionary']:
         raise ValueError("`output_type` must be either 'list' or 'dict'.")
 
+    ### We first want to identify and group all blended lines
+    ### Sort them by observed wavelength to do this
+    lst=[]
+    compnos=[]
+    for ii,comp in enumerate(comp_list):
+        lst.extend(comp._abslines)
+        compnos.extend([ii]*len(comp._abslines))
+    lst=np.array(lst)
+    compnos=np.array(compnos)
+    wv1s=np.array([line.limits.wvlim[0].value for line in lst])
+    sortidxs=np.argsort(wv1s)
+    sort_lst=lst[sortidxs]
+    sort_compnos=compnos[sortidxs] # This will store indices of the lines' parent comps.
+
+    ### Identify the blends
+    blends = []
+    for i in range(len(sortidxs)-1):
+        if i == 0:
+            thisblend = [i]
+        if sort_lst[i].coincident_line(sort_lst[i+1]):
+            thisblend.append(i+1)
+            if i==(len(sortidxs)-2):
+            	blends.append(thisblend)
+        else:
+            blends.append(thisblend)
+            thisblend = [i+1]
+            if i==(len(sortidxs)-2):
+            	blends.append(thisblend)
+
+    ### Associate the lines to their parent components
+    blendnos=[]
+    for blist in blends:
+        blendnos.append(sort_compnos[blist])
+
+    ### Main algorithm to group together all components with blended lines
+    compfound=[]
+    grblends=[]
+    newgroups=[]
+    for i,bn in enumerate(blendnos):
+        if i in grblends: continue
```

Comment:
add inline comment saying that if a component is already in grblends we won't append it again. btw, what does `grblends` stand for? May I suggest to ghange it to `compblends` or something like that.